File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target, nums):
        current_sum = 0
        min_length = float('inf')  # Start with infinity
        start = 0  # Start pointer

        for end in range(len(nums)):
            current_sum += nums[end]  # Add the current number to the current sum

            while current_sum >= target:  # While the sum is at least the target
                min_length = min(min_length, end - start + 1)  # Update the min_length
                current_sum -= nums[start]  # Shrink the window from the left
                start += 1  # Move start to the right

        return min_length if min_length != float('inf') else 0  # Return result

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        //here we store the current sum of the element
        int currentsum = 0;
        int minlength = Integer.MAX_VALUE;
        int start = 0 ;  
        for(int end = 0 ;end < nums.length ; end++){
            currentsum += nums[end];
            while(currentsum >= target){
                minlength = Math.min(minlength , end-start +1);
                currentsum -= nums[start];
                start++ ;
            }
        }
                            return minlength == Integer.MAX_VALUE ? 0 : minlength;

    }
}
    









